### DISCUSSION TITLE
SetOption68 and rotary encoder dimmer for all channels

### DISCUSSION THREAD

**Discussion by lj502** [arendst/Tasmota#17561] [Q&A]
I need to be able to control multiple led strips off on controller, I have a number of magichome controllers flashed with Tasmota that work great and use a rotary encoder for local dimming and power control. On a couple I need to use the setoptionb68 for seperate PWM control, all strips are white, but I would like to be able to dim all the same with a rotary encoder, I used a rule (ON rotary1#pos1 DO Backlog channel1 %value%; channel2 %value%; channel3 %value% ENDON) but after setting option68 to 1 the rotary encoder steps by 1 now instead of 10 and it is painfully slow to use this way. I need the standard 10 step control back with setoption68 set. I'm stuck now
Thanks
John

**Comment by arendst** [comment]
Could ``SetOption43 10`` be an option if ``SetOption98 1``?

**Reply by lj502** [reply]
Thanks, I have so68 on for pwm channel control, so 98 on with rule above, so43 set to 10 or 100 gives the same control rate from encoder, increase or decrease by 1, with 10 the output just stops at 10% so no full brightness, at 100 the strips work fine but the encoder steps by 1 and it is way to slow this way. With all other lights, so68 off, the encoder works as a dimmer with 10 steps to 100, with so68 on that is  lost and it steps 1 number for each click of encoder. Maybe using one value for all pwm channels, not sure if that is possible. I know the dim rate is set at compile to 10, is there a way to use "dimmer" on all channels of a multichannel setup?  I really need hardware control of dimming for this setup to be proper. 
John

**Reply by lj502** [reply]
I noticed using "channel# + or -" in the console increases or decreases by 10 steps, is there a way to use channel+ and channel- tied to a rotary encoder for the up/down control in a rule?
John

**Reply by arendst** [reply]
You should multiply the %value% by ten, store it in a var %var1% and then do the backlog commands using the %var1% instead of %value%

**Reply by sfromis** [reply]
For multiplying the integer by ten, you can skip the multiplication, and just use `%value%0` directly in the command.

**Reply by lj502** [reply]
I thought about it for a while and decided to use one output for all strips, they all need to work the same, so I pulled an old RGB amplifier out of the parts box and added it to drive the strips. I used a single PWM output, keeps it an actual LIGHT for control, and split it to 2 inputs on the amplifier, works perfect and no rules to add, dimmer works as it should and can control a large load now.
Thanks for the suggestions but I was making it harder than it needed to be I think.
John
